lib/auth/trustedcluster.go

```diff
@@ -322,6 +322,23 @@ func (a *Server) addCertAuthorities(trustedCluster types.TrustedCluster, remoteC
 	return nil
 }
 
+func (a *Server) updateUserCARoleMap(ctx context.Context, existingCluster, trustedCluster types.TrustedCluster) error {
+	certAuthority, err := a.GetCertAuthority(ctx, types.CertAuthID{
+		Type:       types.UserCA,
+		DomainName: existingCluster.GetName(),
+	}, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if certAuthority.GetType() == types.UserCA {
+		certAuthority.SetRoleMap(trustedCluster.GetRoleMap())
+	}
+	if err := a.UpsertCertAuthority(certAuthority); err != nil {
```

Comment:
This should be a CAS operation, as the root cluster might have rotated its own user CA between the `GetCertAuthority` and the `UpsertCertAuthority` in this function, causing the local cluster to essentially roll back the remote CA, leading to very confusing behavior (not super critical, as it would eventually resync, but still).